tests/stream_test.c

```diff
@@ -46,6 +52,24 @@ static struct aws_input_stream *s_create_binary_file_stream(struct aws_allocator
     return aws_input_stream_new_from_file(allocator, s_test_file_name);
 }
 
+static struct aws_input_stream *s_create_read_only_file_stream(struct aws_allocator *allocator) {
+    remove(s_test_file_name);
+
+    FILE *file = aws_fopen(s_test_file_name, "w+b");
+    fwrite(s_simple_binary_test, sizeof(uint8_t), sizeof(s_simple_binary_test), file);
+    fclose(file);
+#ifdef _WIN32
+    if (_chmod(s_test_file_name, 0444)) {
+        return NULL;
+    }
+#else
+    if (chmod(s_test_file_name, 0444)) {
```

Comment:
I'm weirded out that this is  `0444` and not `0x444`
But then ... I've never called `chmod()` from C

Maybe just use the enum `S_IRUSR`?